src/components/coins-data-display/CryptoDisplay.component.tsx

```diff
@@ -17,17 +30,87 @@ const CryptoDisplay: React.FC = () => {
     });
   }, []);
 
+  // if coins array changes, update filteredcoins
+  useEffect(() => {
+    setfilteredcoins(coins);
+  }, [coins])
+
   // compares coins to the search input and sets the result to filteredcoins
   const handleSearch = (event: React.ChangeEvent<HTMLInputElement>): void => {
     const lowerCase = event.target.value.toLowerCase();
     const searchedCoins = coins.filter(coin => coin.name.toLowerCase().includes(lowerCase));
+    sortByColumn(column, searchedCoins);
 
     setfilteredcoins(searchedCoins);
   };
 
+  const handleColumn = (event: React.MouseEvent<HTMLButtonElement>, clickedColumn: Column): void => {
+    let filteredcoinsCopy = [...filteredcoins];
+
+    if (column === clickedColumn) {
+      filteredcoinsCopy.reverse();
+    } else {
+      sortByColumn(clickedColumn, filteredcoinsCopy);
+    }
+
+    setfilteredcoins(filteredcoinsCopy);
+    setcolumn(clickedColumn);
+  };
+
+  // sorts the ICoin array by the given column (clickedColumn).
+  const sortByColumn = (clickedColumn: Column, array: Array<ICoin>): void => {
+    switch(clickedColumn) {
+      case Column.Name:
+        array.sort((a, b) => {
+          return a.name.localeCompare(b.name);
+        });
+        break;
+      case Column.Price:
+        array.sort((a, b) => {
+          return b.current_price - a.current_price;
+        });
+        break;
+      case Column.MarketChange:
+        array.sort((a, b) => {
+          return b.market_cap_change_percentage_24h - a.market_cap_change_percentage_24h;
+        });
+        break;
+      case Column.MarketCap:
+        array.sort((a, b) => {
+            return b.market_cap - a.market_cap;
+        });
+    }
+  }
+
   return (
     <div className='coinlist-wrapper'>
       <input className='searchbar' onChange={handleSearch} placeholder="Search for a coin..." ></input>
+      <div id="column-div" className='column-buttons'>
+        <button className={"column-button" + (column === Column.Name ? "-active" : "")} onClick={(e) => handleColumn(e, Column.Name)} >
+          <IconContext.Provider value={{ size: "1.5em" }}>
```

Comment:
Can everything be wrapped inside IconContext.Provider? Or is it mandatory to use it 4 times